### DISCUSSION TITLE
Conversion of Genie to Avalonia from WinForms

### DISCUSSION THREAD

**Discussion by jherig** [GenieClient/Genie4#153] [General]
Now that Genie4 is .Net6, I would propose we move towards using a cross-platform UI, so that Genie4 (or 5 if we bump the number after the uplift) could run on Linux and MacOS natively.

I am interested in doing the majority of the work, if other developers have no objections.  I've all 3 platforms in my house, so can test, and some limited experience with Avalonia and want (need?) more, and think this is a a great project to start with.

What say you all?

**Comment by dylb0t** [comment]
It is funny you should mention this, it is exactly what Djordje told me should be the proper progression. I was saying to him I wanted to write something in GTK  but he said it would make more sense moving to a true cross platform base, s/a Avalonia.
At any rate, I have been working on it for a while, but now AI helped me get something faster. (instead of 1 hour a day, it has greatly improved my productivity)
Here it is: https://github.com/dylb0t/Genie5
I created developer documentation to explain what I did here: https://github.com/dylb0t/Genie5/tree/main/docs
Ultimately I would love to see this moved from my account to GenieClient account if it is accepted by the community.
